### PR TITLE
use `sotp` for TOTP secrets

### PR DESCRIPTION
**Pull request by zowoq**
I'd imagine that eventually we'd want to move to a password manager but this is enough of an improvement over the current workflow that we can put off making that decision.

https://github.com/getsops/sotp

Unlicensed at the moment, I'll add it to nixpkgs once upstream addresses that.

### CODE REVIEW

**Comment by Mic92** [comment]
What shared totp secrets do we currently have?

**Comment by zimbatm** [comment]
the github bot accounts for example, since GitHub requires MFA on all the accounts now

**Review by Mic92 — APPROVED** [review]

File: dev/shell.nix [hunk 1]
```diff
@@ -15,5 +15,22 @@
         python3.pkgs.mkdocs-material
       ];
     };
+    sotp = with pkgs; mkShellNoCC {
```
Comment (zowoq) [inline dev/shell.nix:18]: Don't need this in the default devshell.

File: tasks.py [hunk 2]
```diff
@@ -41,6 +41,14 @@ def deploy(h: DeployHost) -> None:
     g.run_function(deploy)
 
 
+@task
+def sotp(c: Any, acct: str) -> None:
+    """
+    Get TOTP token from sops
+    """
+    c.run(f"nix develop .#sotp -c sotp {acct}")
```
Comment (zimbatm) [inline tasks.py:49]: I think you could get away with making this nix runnable no? `f"nix run .#sotp {acct}"` (and change the above recipe)

Comment (zowoq) [reply tasks.py:49]: Once this package is added to nixpkgs we'd be changing it back to a devshell anyway?

**Pull request merged**